dart/lib/src/sentry_stack_trace_factory.dart

```diff
@@ -147,26 +148,32 @@ class SentryStackTraceFactory {
     final scheme = frame.uri.scheme;
 
     if (scheme.isEmpty) {
-      return true;
+      // Early bail out.
+      return _options.isStackFrameInAppDefault;
     }
+    // The following code depends on the scheme being set.
 
-    for (final include in _options.inAppIncludes) {
-      if (frame.package != null && frame.package == include) {
+    final package = frame.package;
+    if (package != null) {
+      if (_options.inAppIncludes.contains(package)) {
         return true;
       }
-    }
 
-    for (final exclude in _options.inAppExcludes) {
-      if (frame.package != null && frame.package == exclude) {
+      if (_options.inAppExcludes.contains(package)) {
         return false;
       }
     }
 
-    if (frame.isCore ||
-        (frame.uri.scheme == 'package' && frame.package == 'flutter')) {
+    if (frame.isCore) {
+      // This is a Dart frame
+      return false;
+    }
+
+    if (frame.package == 'flutter') {
```

Comment:
could still be a `||` operation, right separating it?

Reply:
Yep, but I think it's worth being more explicit here because it's such an important part.

Reply:
I think a `||` operation is just as clean as 2 ifs, the downside is that this has a cost, well, its probably not even noticeable, computers are designed for this, I don't have strong feelings here so feel free to keep it as it is